src/GitProjectHealth-Model-Importer/GitModelImporter.class.st

```diff
@@ -145,9 +145,15 @@ GitModelImporter >> blockForBranchEquality [
 
 { #category : #equality }
 GitModelImporter >> blockForDiffEquality [
-	^ [ :existing :new |
-			existing diffString size = new diffString size and: [
-				existing diffString = new diffString ] ]
+
+^ [ :existing :new |
+	  | sameSize sameDiff sameFile |
+	  sameSize := existing diffString size = new diffString size.
+	  sameDiff := existing diffString = new diffString.
+	  sameFile := existing new_path = new new_path.
+
+	  sameSize and: (sameDiff and: sameFile).
```

Comment:
```suggestion
	  sameSize and: [ sameDiff and: [ sameFile ] ].
```

We should use `[]` when using `and:` and `or:` for performance reason.
It evaluates only needed part. Here, we should also put inside the equality check (to avoid performing long check if not needed)